core/src/main/scala/kafka/controller/KafkaController.scala

```diff
@@ -1733,16 +1733,31 @@ class KafkaController(val config: KafkaConfig,
       else brokerEpoch
 
     val shouldSkipShutdownSafetyCheck = controllerContext.skipShutdownSafetyCheck.getOrElse(id, -1L) >= actualBrokerEpoch
+    var passedAllShutdown: Boolean = true // For logging purpose
+    if (controllerContext.getLivePreferredControllerIds.size <= config.liMinPreferredControllerCount
+        // Only check if this broker affects live preferred controller set so that for non-preferred-controller clusters it doesn't get blocked
+        && controllerContext.getLivePreferredControllerIds.contains(id)) {
+      passedAllShutdown &= false
```

Comment:
Why do you need this? And why not directly set it to false? Looks like  passedAllShutdown &= false will always set the value to false.

Reply:
That's a good catch!